core/templates/dev/head/domain/collection/CollectionNodeObjectFactory.js

```diff
@@ -17,20 +17,30 @@
  * collection node domain objects.
  */
 
-// TODO(bhenning): Implement validation functions and related tests.
+// TODO(bhenning): This should be provided by the backend.
```

Comment:
In the meantime, you get skew, though. I don't know when we'll turn down GLOBALS, and avoiding skew in the meantime is pretty important IMO.